src/python-common/ceph/rgw/rgwam_core.py

```diff
@@ -541,40 +609,28 @@ def realm_new_zone_creds(self, realm_name, endpoints, sys_uid):
 
     def realm_rm_zone_creds(self, realm_token_b64):
         if not realm_token_b64:
-            print('missing realm token')
-            return False
-
-        realm_token_b = base64.b64decode(realm_token_b64)
-        realm_token_s = realm_token_b.decode('utf-8')
-
-        realm_token = json.loads(realm_token_s)
-
-        access_key = realm_token['access_key']
-        realm_id = realm_token['realm_id']
+            raise RGWAMException('missing realm token')
 
+        realm_token = RealmToken.from_base64_str(realm_token_b64)
         try:
-            period_info = self.period_op().get(EntityID(realm_id))
+            period_info = self.period_op().get(EntityID(realm_token.realm_id))
         except RGWAMException as e:
             raise RGWAMException('failed to fetch period info', e)
 
         period = RGWPeriod(period_info)
-
         master_zg = EntityID(period.master_zonegroup)
         master_zone = EntityID(period.master_zone)
-
         logging.info('Period: ' + period.id)
         logging.info('Master zone: ' + period.master_zone)
-
         try:
             zone_info = self.zone_op().get(zone=master_zone)
         except RGWAMException as e:
             raise RGWAMException('failed to access master zone', e)
 
-        zone_id = zone_info['id']
-
-        if period.master_zone != zone_id:
+        if period.master_zone != zone_info['id']:
             return (-errno.EINVAL, '', 'Command needs to run on master zone')
 
+        access_key = realm_token.access_key
         try:
             user_info = self.user_op().info(master_zone, master_zg, access_key=access_key)
         except RGWAMException as e:
```

Comment:
This exception says it is failing to create a system user when it's doing a `user info` command not a `user create`.

Reply:
Fixed